### PR TITLE
Live Preview: Improve loading state when starting Preview & Customize

### PR DESCRIPTION
**Pull request by fushar**
Related to:

- https://github.com/Automattic/wp-calypso/issues/84401

## Proposed Changes

This PR improves the loading state of Preview & Customize feature, in the theme details page and theme showcase cards. It's split into 2 commits if you want to review them separately.

### Theme Details Page

We're updating the button to loading state when clicked:

|Before|After|
|-|-|
|image|image|

Note: I didn't change the button text when in loading state (such as to `Preparing Preview & Customize...`). I think it's weird (and too wordy) because we are redirecting to another page anyway, and user won't see this button finishes loading.

---

### Theme Showcase Card

We're reusing the loading state when a theme is being installed:

|Before|After|
|-|-|
|image|image|

### Theme Details Page

1. Go to `/theme/twentytwentyfour/<siteSlug>`
2. Click "Preview & Customize" button
3. Verify that the button changes to loading state
4. Also try for Community themes (in a Business plan)

### Theme Showcase Card

1. Go to `/themes/<siteSlug>`
2. Click the "three-dot" menu of any free theme
3. Click "Preview & Customize" menu
5. Verify that the theme card changes to loading state
6. Bonus (regression test): use an Atomic site, try installing a theme, make sure that the card still changes to a loading state

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by arthur791004 — APPROVED** [review]
Tested and looks good 👍

File: client/my-sites/theme/live-preview-button/index.tsx [hunk 1]
```diff
@@ -19,17 +19,24 @@ export const LivePreviewButton: FC< Props > = ( { themeId, siteId } ) => {
...
 		getIsLivePreviewSupported( state, themeId, siteId )
 	);
 	if ( ! isLivePreviewSupported ) {
 		return null;
 	}
 
+	const handleClick = () => {
+		setIsLoading( true );
+		dispatch( livePreview( themeId, siteId, 'detail' ) );
+	};
```
Comment (okmttdhr) [inline client/my-sites/theme/live-preview-button/index.tsx:34]: Just to hear your opinion, but should we implement error handling and terminate the loading state at this point? For example, in cases where the theme installation fails.

Comment (fushar) [reply client/my-sites/theme/live-preview-button/index.tsx:34]: I try to time-box this but it is complicated as there is currently nothing in the codebase that listens to the `THEME_INSTALL_FAILURE` action 😭 

We can follow-up in a future PR, what do you think?

**Comment by fushar** [comment]
@arthur791004 / @okmttdhr : I found a bug when live-previewing ~~Community themes on Business plan sites~~, could you retest this with that scenario? 🙇

**Comment by fushar** [comment]
@arthur791004 / @okmttdhr: Sorry, I mean on Atomic sites! (when themes need to be installed first) My bad

File: client/my-sites/theme/live-preview-button/index.tsx (hunk 1 above)
Comment (okmttdhr) [reply client/my-sites/theme/live-preview-button/index.tsx:34]: >I try to time-box this but it is complicated as there is currently nothing in the codebase that listens to the THEME_INSTALL_FAILURE action 😭

>We can follow-up in a future PR, what do you think?

It makes sense to me 🙆‍♂️ Thanks for taking a look at it!

**Comment by miksansegundo** [comment]
> Note: I didn't change the button text when in loading state (such as to Preparing Preview & Customize...). I think it's weird (and too wordy) because we are redirecting to another page anyway, and user won't see this button finishes loading.

It's interesting that the [Material design style guide](https://m3.material.io/foundations/content-design/style-guide/grammar-and-punctuation#ba2fc0de-152d-4562-b43d-abd6a9378a94) suggests avoiding the use of ellipses in buttons.

Screenshot 2566-12-04 at 15 27 01

**Review by arthur791004 — APPROVED** [review]
Tested on my AT site again, and it still works as expected.

✅ Click the "Preview & Customize" button in the Theme Showcase
✅ Click the "Preview & Customize" button in the Theme Details page
